src/Features/Core/Portable/SemanticSearch/QueryExecutionContext.cs

```diff
@@ -130,30 +130,49 @@ private async ValueTask InvokeAsync(Project project, Compilation compilation, ob
 
             try
             {
-                var symbols = (IEnumerable<ISymbol?>?)method.Invoke(null, [entity]) ?? [];
+                var symbols = method.Invoke(null, [entity]);
+                if (symbols != null)
+                {
+                    if (isAsync)
+                    {
+                        await foreach (var symbol in ((IAsyncEnumerable<ISymbol?>)symbols).WithCancellation(cancellationToken).ConfigureAwait(false))
+                        {
+                            await ObserveSymbolAsync(symbol).ConfigureAwait(false);
+                        }
+                    }
+                    else
+                    {
+                        foreach (var symbol in (IEnumerable<ISymbol?>)symbols)
+                        {
```

Comment:
consider checking CT here.  same with foreach loop above.

Reply:
It is checked in ObserveSymbolAsync. Refactored this.